source/common/upstream/load_balancer_impl.cc

```diff
@@ -125,7 +125,9 @@ bool LoadBalancerUtility::isGlobalPanic(const HostSet& host_set, ClusterStats& s
                                         Runtime::Loader& runtime) {
   uint64_t global_panic_threshold =
       std::min<uint64_t>(100, runtime.snapshot().getInteger(RuntimePanicThreshold, 50));
-  double healthy_percent = 100.0 * host_set.healthyHosts().size() / host_set.hosts().size();
+  double healthy_percent = host_set.hosts().size() == 0
```

Comment:
Ah OK. Then perhaps just remove the empty() checks from the two hostsToUse() callsites since they are no longer needed? Then we don't do double checks and have the logic closer to where it needs to be?